tests/test_tokenizer.py

```diff
@@ -51,11 +50,23 @@ def test_tokenizer_against_hf(config):
 
     assert ours.vocab_size == theirs.vocab_size
     assert ours.vocab_size == config.vocab_size
-    assert ours.bos_id == theirs.bos_token_id
-    assert ours.eos_id == theirs.eos_token_id
+
+    if config.name.startswith("falcon") or config.name.startswith("stablecode"):
+        # even though their config defines it, it's set as None in HF
+        assert isinstance(ours.bos_id, int)
+        assert theirs.bos_token_id is None
+    else:
+        assert ours.bos_id == theirs.bos_token_id
+
+    if config.name.startswith("stablecode"):
+        # even though their config defines it, it's set as None in HF
+        assert ours.eos_id == 0
+        assert theirs.eos_token_id is None
+    else:
+        assert ours.eos_id == theirs.eos_token_id
```

Comment:
@nkasmanoff I added these special checks. As you observed, seems like the HF tokenizer is a bit broken for falcon and stablecode :sweat_smile: